<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QDynDialogRobotTable</class>
 <widget class="QDialog" name="QDynDialogRobotTable">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>803</width>
    <height>291</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <property name="sizeGripEnabled">
   <bool>true</bool>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout" columnstretch="0,0,0,0,0">
   <property name="sizeConstraint">
    <enum>QLayout::SetDefaultConstraint</enum>
   </property>
   <item row="1" column="1" colspan="4">
    <widget class="QTableWidget" name="robotTable">
     <property name="contextMenuPolicy">
      <enum>Qt::CustomContextMenu</enum>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::AnyKeyPressed|QAbstractItemView::EditKeyPressed|QAbstractItemView::SelectedClicked</set>
     </property>
     <property name="alternatingRowColors">
      <bool>true</bool>
     </property>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>90</number>
     </attribute>
     <attribute name="horizontalHeaderMinimumSectionSize">
      <number>90</number>
     </attribute>
     <column>
      <property name="text">
       <string>Name</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Parent</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Joint</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>a [m]</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>α [°]</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>d [m]</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>θ [°]</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>θ+ [°]</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>qMin</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>qMax</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Mass [Kg]</string>
      </property>
     </column>
    </widget>
   </item>
   <item row="2" column="0" rowspan="2" colspan="5">
    <widget class="Line" name="line_2">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Robot's Name:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_2">
     <property name="minimumSize">
      <size>
       <width>90</width>
       <height>120</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>90</width>
       <height>300</height>
      </size>
     </property>
     <property name="text">
      <string>Robot's DH Table:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="4" column="4">
    <widget class="QPushButton" name="closeDialog">
     <property name="minimumSize">
      <size>
       <width>75</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>75</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="text">
      <string>Close</string>
     </property>
     <property name="icon">
      <iconset resource="../Resources.qrc">
       <normaloff>:/resources/icons/cancel.png</normaloff>:/resources/icons/cancel.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="autoDefault">
      <bool>true</bool>
     </property>
     <property name="default">
      <bool>false</bool>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="0" column="1" colspan="4">
    <widget class="QLineEdit" name="robotName"/>
   </item>
   <item row="4" column="3">
    <widget class="QPushButton" name="save">
     <property name="minimumSize">
      <size>
       <width>90</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>90</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="toolTip">
      <string>Save the current Denavit-Hartenber parameters table into a XML file.</string>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="text">
      <string>Save as...</string>
     </property>
     <property name="icon">
      <iconset resource="../Resources.qrc">
       <normaloff>:/resources/icons/save_as.png</normaloff>:/resources/icons/save_as.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="autoDefault">
      <bool>true</bool>
     </property>
     <property name="default">
      <bool>false</bool>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
    </widget>
   </item>
  </layout>
  <action name="actionAddLink">
   <property name="icon">
    <iconset resource="../Resources.qrc">
     <normaloff>:/resources/icons/link_add.png</normaloff>:/resources/icons/link_add.png</iconset>
   </property>
   <property name="text">
    <string>Add Link at bottom...</string>
   </property>
   <property name="toolTip">
    <string>Add Link to the current D-H Table</string>
   </property>
  </action>
  <action name="actionDeleteLink">
   <property name="icon">
    <iconset resource="../Resources.qrc">
     <normaloff>:/resources/icons/delete.png</normaloff>:/resources/icons/delete.png</iconset>
   </property>
   <property name="text">
    <string>Delete this Link</string>
   </property>
   <property name="toolTip">
    <string>Delete selected Link</string>
   </property>
  </action>
  <action name="actionClearTable">
   <property name="icon">
    <iconset resource="../Resources.qrc">
     <normaloff>:/resources/icons/table_clear.png</normaloff>:/resources/icons/table_clear.png</iconset>
   </property>
   <property name="text">
    <string>Clear Table</string>
   </property>
   <property name="toolTip">
    <string>Clear the current D-H Table</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="../Resources.qrc"/>
 </resources>
 <connections/>
</ui>
